File: frameworks/dotnet/wolverine-http/UnitTests/StreamTests.cs
namespace UnitTests;

[Collection(nameof(WolverineApp))]
public sealed class StreamTests(WolverineApp app)
{
    // rb:test stream.ndjson
    [Fact]
    [Trait("corpus", "stream.ndjson")]
    public async Task Each_row_is_a_line_and_the_length_is_never_sent()
    {
        IScenarioResult result = await app.Host.Scenario(s =>
        {
            s.Get.Url("/stream/items");
            s.ContentTypeShouldBe("application/x-ndjson");
        });

        Assert.Null(result.Context.Response.ContentLength);
        string[] lines = result.ReadAsText().TrimEnd('\n').Split('\n');
        JsonArray rows = Expected.Json("items.medium.json")["items"]!.AsArray();
        Assert.Equal(rows.Count, lines.Length);
        Assert.All(lines.Zip(rows), pair => Assert.True(JsonNode.DeepEquals(pair.Second, JsonNode.Parse(pair.First))));
    }
}

File: frameworks/java/helidon-se/UnitTests/src/test/java/unittests/StreamTests.java
package unittests;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.net.http.HttpResponse;

import com.fasterxml.jackson.databind.JsonNode;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

class StreamTests extends HelidonApp {

    // rb:test stream.ndjson
    @Test
    @Tag("stream.ndjson")
    void eachRowIsALineAndTheLengthIsNeverSent() throws Exception {
        HttpResponse<byte[]> response = get("/stream/items");

        assertTrue(Answer.header(response, "content-type").startsWith("application/x-ndjson"));
        assertNull(Answer.header(response, "content-length"));
        String[] lines = Answer.text(response).stripTrailing().split("\n");
        JsonNode rows = Expected.json("items.medium.json").get("items");
        assertEquals(rows.size(), lines.length);
        for (int i = 0; i < lines.length; i++) {
            assertEquals(rows.get(i), Expected.JSON.readTree(lines[i]));
        }
    }
}
